src/main/java/org/spongepowered/api/CatalogType.java

```diff
@@ -25,8 +25,52 @@
 package org.spongepowered.api;
 
 /**
- * Represents a type of a catalog that can be used to identify types
- * without using an {@link Enum}.
+ * Represents a type of a catalog that can be used to identify types without
+ * using an {@link Enum}.
+ *
+ * <p>Implementing classes must meet the following requirements for all its
+ * members including those which are not listed in the classes listed in the
+ * {@link org.spongepowered.api.util.annotation.CatalogedBy CatalogedBy}
+ * annotation classes:
+ * <ul>
+ * <li>Every pair (a, b) that matches ...
+ * <ul>
+ * <li>a.equals(b) must also match a == b</li>
+ * <li>a.getId().equalsIgnoreCase(b.getId()) must also match a == b</li>
+ * <li>a.getName().equalsIgnoreCase(b.getName()) must also match a == b</li>
+ * </ul>
+ * </li>
+ * <li>The {@link GameRegistry#getType(Class, String)} method must serve all
+ * values independently from the case of the real id value.
+ * <ul>
+ * <li>It may but does not need to return the values for "well known aliases"
+ * </li>
+ * </ul>
+ * </li>
+ * <li>The value of a.getId() identifies the reference uniquely and must remain
```

Comment:
These two are just rehashing the javadocs on the methods, better to not repeat yourself